toffy/normalize.py

```diff
@@ -216,26 +232,37 @@ def combine_run_metrics(run_dir, substring):
     metrics.to_csv(os.path.join(run_dir, substring + '_combined.csv'), index=False)
 
 
-def combine_tuning_curve_metrics(dir_list):
+def combine_tuning_curve_metrics(dir_list, count_range):
     """Combines metrics together into a single dataframe for fitting a turning curve
 
     Args:
         dir_list (list): list of directories to pull metrics from
+        count_range (tuple): range of appropriate mass count values for the FOVs
 
     Returns:
         pd.DataFrame: dataframe containing aggregates metrics"""
 
     # create list to hold all extracted data
-    all_dirs = []
-
+    all_dirs, excluded_fovs = [], []
     # loop through each run folder
     for dir in dir_list:
-
-        # combine tables together
         pulse_heights = pd.read_csv(os.path.join(dir, 'fov-1-scan-1_pulse_heights.csv'))
         channel_counts = pd.read_csv(os.path.join(dir, 'fov-1-scan-1_channel_counts.csv'))
-        combined = pulse_heights.merge(channel_counts, 'outer', on=['fov', 'mass'])
 
+        # check for extreme count values
+        extreme_val = False
+        if count_range:
+            for count in channel_counts.channel_count:
```

Comment:
Rather than a loop over the counts, I would use numpy to compare all the elements at once, and see if any are true